File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int r = 1;
        int sum = 0;
        while(r< prices.length){
          if(prices[r] > prices[l]){
              int profit = prices[r] - prices[l];
              System.out.println(profit);
              sum+=profit;
              
              l++;
              
          }
          else{
              l=r;
          }
          r++;
        }

        return sum;
    }
}

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxL = 0
        for i in range(1,len(prices)):
            if prices[i]> prices[i-1]:
                maxL += max(0,prices[i]-prices[i-1])
        return maxL